zabbix/datadog_checks/zabbix/check.py

```diff
@@ -27,23 +27,32 @@ def request(self, zabbix_api, req_data):
         return res.json()
 
     def login(self, zabbix_user, zabbix_pass, zabbix_api):
-        req_data = json.dumps(
-            {
-                'jsonrpc': '2.0',
-                'method': 'user.login',
-                'params': {'user': zabbix_user, 'password': zabbix_pass},
-                'id': 1,
-            }
-        )
         self.log.debug("Logging in with params user=%s api=%s", zabbix_user, zabbix_api)
 
-        response = self.request(zabbix_api, req_data)
-        token = response.get('result')
-        if token is None:
-            raise Exception(
-                'Unable to login with params user={} api={}: {}'.format(zabbix_user, zabbix_api, response.get('error'))
+        # Try both 'user' (older versions) and 'username' (Zabbix 5.4+) for backward compatibility
```

Comment:
nit- I'd cache the working method somehow. So only try potentially try both at the first run and then cache the result and only do that in the subsequent calls.